Use onreadychangestate event for script initialization

johnp opened the pull request:
Due to an [issue](https://github.com/codebicycle/videospeed/issues/4) with the Firefox fork of this extension I've had a look at this and it would probably be nicer to use the event instead of constantly polling.
This is untested, but literally a copy-paste from [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState) so it should work. The [`load`](https://developer.mozilla.org/en-US/docs/Web/Events/load) event would be semantically equivalent.

igrigorik commented:
In theory this should work.. In practice, it doesn't in many cases. :(

Take a look at the commit history for that code.

johnp commented:
I checked the commit history back until "initial import" or so and couldn't find anything, but if you say so...
Is the `load` event any better?
Could it be that the event listener gets attached after the event has already fired and adding a manual check for readyState before that helps?

igrigorik commented:
@johnp thanks for digging into this. Merged your patch locally, going to try to browse around with it and see if I run into any issues.

johnp commented:
I've changed the ordering a bit (add the window.onload event handler first because `document` may be null if injected early), added a missing null-check for ` document` and squashed the commits. WFM in the Firefox port on Nightly.

igrigorik commented:
Been running with the earlier version for a while -- everything seems a-ok. 

Merging, thanks for your help and patience! :)
